src/bootstrap/bin/main.rs

```diff
@@ -9,10 +9,26 @@ use std::env;
 
 use bootstrap::{Build, Config, Subcommand, VERSION};
 
-fn main() {
+fn main() -> std::io::Result<()> {
     let args = env::args().skip(1).collect::<Vec<_>>();
     let config = Config::parse(&args);
 
+    let mut build_lock;
+    let _build_lock_guard;
+    if cfg!(any(unix, windows)) {
+        build_lock = fd_lock::RwLock::new(std::fs::File::create(config.out.join("lock"))?);
+        _build_lock_guard = match build_lock.try_write() {
+            Ok(lock) => lock,
+            err => {
+                println!("warning: build directory locked, waiting for lock");
+                drop(err);
+                build_lock.write()?
```

Comment:
The error handling convention in bootstrap is the `t!` macro, which exits with a helpful message on error.
```suggestion
        build_lock = fd_lock::RwLock::new(t!(std::fs::File::create(config.out.join("lock"))));
        _build_lock_guard = match build_lock.try_write() {
            Ok(lock) => lock,
            err => {
                println!("warning: build directory locked, waiting for lock");
                drop(err);
                t!(build_lock.write())
```

Reply:
`t` appears to be crate-private to the `bootstrap` library crate; I can move it to `pub`, if you'd like it used in the binary too.